lib/elixir/lib/calendar.ex

```diff
@@ -151,10 +228,10 @@ defmodule Date do
       true
 
   """
-  @spec utc_today() :: t
-  def utc_today() do
-    {:ok, {year, month, day}, _, _} = Calendar.ISO.from_unix(:os.system_time, :native)
-    %Date{year: year, month: month, day: day}
+  @spec utc_today(Calendar.calendar) :: t
+  def utc_today(calendar \\ Calendar.ISO) do
```

Comment:
We cannot do literally the same that we did with `Time.utc_now/1` as Times can always be converted as-is, but Dates and NaiveDateTimes cannot; to ensure the 'UNIX -> Date' and 'UNIX -> NaiveDateTime' conversions happen properly, we need to use DateTime's implementation of convert.

We could duplicate the code of `DateTime.to_rata_die` and `DateTime.from_rata_die` instead of doing the struct conversions, which might be a little more efficient but also harder to read.